include/crocoddyl/multibody/costs/com-position.hxx

```diff
@@ -36,39 +36,41 @@ CostModelCoMPositionTpl<Scalar>::CostModelCoMPositionTpl(boost::shared_ptr<State
 template <typename Scalar>
 CostModelCoMPositionTpl<Scalar>::CostModelCoMPositionTpl(boost::shared_ptr<StateMultibody> state, const Vector3s& cref,
                                                          const std::size_t nu)
-    : Base(state, 3, nu), cref_(cref) {}
+    : Base(state, boost::make_shared<ResidualModelCoMPosition>(state, cref, nu)), cref_(cref) {}
 
 template <typename Scalar>
 CostModelCoMPositionTpl<Scalar>::CostModelCoMPositionTpl(boost::shared_ptr<StateMultibody> state, const Vector3s& cref)
-    : Base(state, 3), cref_(cref) {}
+    : Base(state, boost::make_shared<ResidualModelCoMPosition>(state, cref)), cref_(cref) {}
 
 template <typename Scalar>
 CostModelCoMPositionTpl<Scalar>::~CostModelCoMPositionTpl() {}
 
 template <typename Scalar>
 void CostModelCoMPositionTpl<Scalar>::calc(const boost::shared_ptr<CostDataAbstract>& data,
```

Comment:
If we move the implementation of costmodelresidual to costmodelbase, we can get rid of the most of the duplication of code in `calc`, `calcdiff`

Reply:
I will mark this conversation as resolved because we agree to keep current implementation (which it will be deprecated soon).